src/ui/components/search/filtersearchcomponent.js

```diff
@@ -85,6 +85,12 @@ export default class FilterSearchComponent extends Component {
      */
     this.redirectUrl = config.redirectUrl || null;
 
+    /**
+     * Whether to execute a search when the page is loaded
+     * @type {boolean}
+     */
+    this.searchOnLoad = config.searchOnLoad;
```

Comment:
I think we need the defaulting behavior to be more complex. If they haven't set searchOnLoad explicitly, we will default to false unless they've set config.filter explicitly. In that case, searchOnLoad should be true. This will support the case where someone has config.query and config.filter set and is expecting a network call in onCreate.